🐛 BUG: Build Requests `account_id` in Wrangler.toml but fails with `account_id`

### Which Cloudflare product(s) does this pertain to?

Workers Runtime

### What version(s) of the tool(s) are you using?

3.70.0 [wrangler]

### What version of Node are you using?

_No response_

### What operating system and version are you using?

Cloudflare Dashboard

### Describe the Bug

https://dash.cloudflare.com/ab7347813b21ea6079df4cb18c4bb9e8/pages/view/exchange-traded-fund/952e6991-3192-47dd-9121-12b7bbce5d01
```  
00:30:22.332    ✘ [ERROR] More than one account available but unable to select one in non-interactive mode.
00:30:22.332    
00:30:22.332      Please set the appropriate `account_id` in your `wrangler.toml` file.
00:30:22.332      Available accounts are (`<name>`: `<account_id>`):
```

https://dash.cloudflare.com/ab7347813b21ea6079df4cb18c4bb9e8/pages/view/exchange-traded-fund/caf73d0b-3b79-4c19-9ce6-e83e33dac0f4
```
00:31:50.811    Found wrangler.toml file. Reading build configuration...
00:31:50.825    
00:31:50.930    ✘ [ERROR] Running configuration file validation for Pages:
00:31:50.930    
00:31:50.931        - Configuration file for Pages projects does not support "account_id"
```

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

--- a/.changeset/twenty-ears-rescue.md
+++ b/.changeset/twenty-ears-rescue.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+improve multi account error message in non-interactive mode

--- a/packages/wrangler/src/__tests__/deploy.test.ts
+++ b/packages/wrangler/src/__tests__/deploy.test.ts
@@ -619,7 +619,7 @@ describe("deploy", () => {
 				await expect(runWrangler("deploy index.js")).rejects
 					.toMatchInlineSnapshot(`
 					[Error: More than one account available but unable to select one in non-interactive mode.
-					Please set the appropriate \`account_id\` in your Wrangler configuration file.
+					Please set the appropriate \`account_id\` in your Wrangler configuration file or assign it to the \`CLOUDFLARE_ACCOUNT_ID\` environment variable.
 					Available accounts are (\`<name>\`: \`<account_id>\`):
 					  \`enterprise\`: \`1701\`
 					  \`enterprise-nx\`: \`nx01\`]

--- a/packages/wrangler/src/__tests__/kv.test.ts
+++ b/packages/wrangler/src/__tests__/kv.test.ts
@@ -1275,7 +1275,7 @@ describe("wrangler", () => {
 					await expect(runWrangler("kv key get key --namespace-id=xxxx"))
 						.rejects.toThrowErrorMatchingInlineSnapshot(`
 						[Error: More than one account available but unable to select one in non-interactive mode.
-						Please set the appropriate \`account_id\` in your Wrangler configuration file.
+						Please set the appropriate \`account_id\` in your Wrangler configuration file or assign it to the \`CLOUDFLARE_ACCOUNT_ID\` environment variable.
 						Available accounts are (\`<name>\`: \`<account_id>\`):
 						  \`one\`: \`1\`
 						  \`two\`: \`2\`]
@@ -1291,7 +1291,7 @@ describe("wrangler", () => {
 					await expect(runWrangler("kv key get key --namespace-id=xxxx"))
 						.rejects.toThrowErrorMatchingInlineSnapshot(`
 						[Error: More than one account available but unable to select one in non-interactive mode.
-						Please set the appropriate \`account_id\` in your Wrangler configuration file.
+						Please set the appropriate \`account_id\` in your Wrangler configuration file or assign it to the \`CLOUDFLARE_ACCOUNT_ID\` environment variable.
 						Available accounts are (\`<name>\`: \`<account_id>\`):
 						  \`one\`: \`1\`
 						  \`two\`: \`2\`]
@@ -1332,7 +1332,7 @@ describe("wrangler", () => {
 					await expect(runWrangler("kv key get key --namespace-id=xxxx"))
 						.rejects.toThrowErrorMatchingInlineSnapshot(`
 						[Error: More than one account available but unable to select one in non-interactive mode.
-						Please set the appropriate \`account_id\` in your Wrangler configuration file.
+						Please set the appropriate \`account_id\` in your Wrangler configuration file or assign it to the \`CLOUDFLARE_ACCOUNT_ID\` environment variable.
 						Available accounts are (\`<name>\`: \`<account_id>\`):
 						  \`one\`: \`1\`
 						  \`two\`: \`2\`]

--- a/packages/wrangler/src/__tests__/pages/secret.test.ts
+++ b/packages/wrangler/src/__tests__/pages/secret.test.ts
@@ -274,7 +274,7 @@ describe("wrangler pages secret", () => {
 						runWrangler("pages secret put the-key --project some-project-name")
 					).rejects.toThrowErrorMatchingInlineSnapshot(`
 						[Error: More than one account available but unable to select one in non-interactive mode.
-						Please set the appropriate \`account_id\` in your Wrangler configuration file.
+						Please set the appropriate \`account_id\` in your Wrangler configuration file or assign it to the \`CLOUDFLARE_ACCOUNT_ID\` environment variable.
 						Available accounts are (\`<name>\`: \`<account_id>\`):
 						  \`account-name-1\`: \`account-id-1\`
 						  \`account-name-2\`: \`account-id-2\`

--- a/packages/wrangler/src/__tests__/secret.test.ts
+++ b/packages/wrangler/src/__tests__/secret.test.ts
@@ -413,7 +413,7 @@ describe("wrangler secret", () => {
 					await expect(runWrangler("secret put the-key --name script-name"))
 						.rejects.toThrowErrorMatchingInlineSnapshot(`
 						[Error: More than one account available but unable to select one in non-interactive mode.
-						Please set the appropriate \`account_id\` in your Wrangler configuration file.
+						Please set the appropriate \`account_id\` in your Wrangler configuration file or assign it to the \`CLOUDFLARE_ACCOUNT_ID\` environment variable.
 						Available accounts are (\`<name>\`: \`<account_id>\`):
 						  \`account-name-1\`: \`account-id-1\`
 						  \`account-name-2\`: \`account-id-2\`

--- a/packages/wrangler/src/user/user.ts
+++ b/packages/wrangler/src/user/user.ts
@@ -1204,7 +1204,7 @@ export async function getAccountId(): Promise<string> {
 		if (e instanceof NoDefaultValueProvided) {
 			throw new UserError(
 				`More than one account available but unable to select one in non-interactive mode.
-Please set the appropriate \`account_id\` in your ${configFileName(undefined)} file.
+Please set the appropriate \`account_id\` in your ${configFileName(undefined)} file or assign it to the \`CLOUDFLARE_ACCOUNT_ID\` environment variable.
 Available accounts are (\`<name>\`: \`<account_id>\`):
 ${accounts
 	.map((account) => `  \`${account.name}\`: \`${account.id}\``)

